pkg/binding/bindingpolicy-resolution.go

```diff
@@ -49,13 +49,25 @@ type bindingPolicyResolution struct {
 	requiresSingletonReportedState bool
 }
 
-// objectData stores the UID, resource version, create-only bit,
+// ObjectData stores the UID, resource version, create-only bit,
 // and statuscollectors for an object.
-type objectData struct {
-	UID              string
-	ResourceVersion  string
-	CreateOnly       bool
+type ObjectData struct {
+	// UID is the UID of the object.
+	UID string
+	// ResourceVersion is the resource version of the object.
+	ResourceVersion string
+	// CreateOnly is a boolean that indicates whether the object was selected
+	// by at least one downsync clause with the createOnly bit set.
+	// If enabled, it means that the object should only be created and not
+	// maintained
+	CreateOnly bool
+	// StatusCollectors is a set of status collector names that are associated
 	StatusCollectors sets.Set[string]
+	// stalenessThresholdSecs dictates the staleness threshold for determining
+	// when the object's collected status is considered stale.
```

Comment:
_collected_ status considered stale? As far as I understand, the propagation augmentation's `stale` bit is about whether what the core knows about status from the WEC is "stale", and this is _input_ to the aggregation function. BTW: since this input can change with the passage of time, what gets the aggregation re-evaluated when this input changes?

Reply:
I'm not sure I understand the difference in the first part.

Regarding the 2nd, currently what causes for reevaluation is any time a workstatus is evaluated, which can occur on the three event streams the resolver is affected by (changes in bindings, changes in statuscollectors and changes in workstatuses).

I'm thinking about using delayed queuing to trigger re-evaluation, where once one of the above triggers an evaluation, the delay is calculated to be the time left for status to become stale, if it isn't.

Reply:
Regarding the first part, another way to put it is to ask exactly what is meant by _collected status_. That is not exactly a term that is used elsewhere in this API. If "collected status" means the result of applying StatusCollectors, that is one thing. If "collected status" means the core's copy of (the object's reported state and the klusterlet's status wrt the object) from the WEC, that is a different thing. So the comment should be updated to be clear. I hope that the clarification says that the age in question is the age of the core's copy of (the object's reported state and the klusterlet's status wrt the object) from the WEC.

Reply:
This is moot now.